lms/djangoapps/grades/new/course_grade.py

```diff
@@ -47,6 +81,8 @@ def locations_to_scores(self):
         """
         Returns a dict of problem scores keyed by their locations.
         """
+        if not self.chapter_grades:
```

Comment:
Make `chapter_grades` a property that is lazily computed when needed.  The implementation of it should simply call the existing `compute_and_update` method, which does the "bulk" operations that we want.  No need to introduce another implementation of the same - so you can remove `_populate_course_structure_grades`.